src/parser/jobs/mnk/modules/Weaving.ts

```diff
@@ -1,15 +1,15 @@
 import ACTIONS from 'data/ACTIONS'
-import CoreWeaving from 'parser/core/modules/Weaving'
+import {CastEvent} from 'fflogs'
+import CoreWeaving, {WeaveInfo} from 'parser/core/modules/Weaving'
 
 const SSS_MAX_WEAVES = 4
 
 export default class Weaving extends CoreWeaving {
-	isBadWeave(weave: TODO /*, maxWeaves*/) {
-		if (weave.leadingGcdEvent &&
-			weave.leadingGcdEvent.ability &&
-			weave.leadingGcdEvent.ability.guid === ACTIONS.SIX_SIDED_STAR.id) {
-				return weave.weaves.length > SSS_MAX_WEAVES
-			}
+	isBadWeave(weave: WeaveInfo /*, maxWeaves*/) {
+		const gcd = weave.leadingGcdEvent as CastEvent
+		if (gcd?.ability && gcd?.ability?.guid === ACTIONS.SIX_SIDED_STAR.id) {
```

Comment:
Stylistically, it would be better to add an isCastEvent type discriminator (similar to the isDamageEvent discriminator in fflogs.ts) and import that to keep TS happy, rather than force-casting the event to a CastEvent -- so then the next check could be something like 
```
if (isCastEvent(gcd) && gcd?.ability?.guid === ACTIONS.SIX_SIDED_STAR.id) 
```
In either case, you can collapse the ?. checks into a single check here, it will bail to null and not equal to SSS the first property it finds that is missing.